core/include/gnuradio-4.0/Block.hpp

```diff
@@ -994,6 +1054,38 @@ struct Block : protected std::tuple<Arguments...> {
         this->state.notify_all();
     }
 
+    constexpr void
+    processScheduledMessages() {
+        // fmt::print("processScheduledMessages for {}\n", unique_name);
+        auto processPort = [this]<PortLike TPort>(TPort &inPort) {
+            const auto available = inPort.streamReader().available();
+            if constexpr (traits::block::can_processMessagesForPort<Derived, TPort>) {
+                if (available > 0) {
+                    self().processMessages(inPort, inPort.streamReader().get(available));
+                }
+            }
+            if (available > 0) {
+                if (auto consumed = inPort.streamReader().consume(available); !consumed) {
+                    throw fmt::format("Could not consume the messages from the message port");
+                }
+            }
+        };
+        processPort(msgIn);
+        for_each_port(processPort, inputPorts<traits::port::kind::Message>(&self()));
+    }
+
+    template<typename TMessage>
+    void
+    emitMessage(TMessage &&message) {
+        emitMessage(msgOut, std::forward<TMessage>(message));
+    }
```

Comment:
Please remove this... can be derived from the one below which makes it also a bit more clear which port the message is being emitted.